WordPress/src/test/java/org/wordpress/android/ui/jetpack/scan/builders/ScanStateListItemsBuilderTest.kt

```diff
@@ -146,43 +157,324 @@ class ScanStateListItemsBuilderTest : BaseUnitTest() {
+            val threatItemState = createDummyThreatItemState(threatModel)
+            val subHeader = DescriptionState(UiStringText("sub header"))
+            whenever(threatItemBuilder.buildThreatItem(threatModel)).thenReturn(threatItemState)
+            whenever(threatDetailsListItemsBuilder.buildFixableThreatDescription(any())).thenReturn(subHeader)
+            whenever(scanStore.getThreatModelByThreatId(any())).thenReturn(threatModel)
+
+            val scanStateItems = buildScanStateItems(
+                model = scanStateModelWithThreats,
+                fixingThreatIds = listOf(threatModel.baseThreatModel.id)
+            )
 
-        val scanStateItems = buildScanStateItems(
-            model = scanStateModelWithThreats,
-            fixingThreatIds = listOf(threatModel.baseThreatModel.id)
+            assertThat(scanStateItems.filterIsInstance(ThreatItemState::class.java).first().subHeader)
+                .isEqualTo(subHeader.text)
+        }
+
+    /* IDLE - THREATS FOUND STATE */
+
+    @Test
+    fun `given idle state with threats, when items are built, items contain shield warning icon with error color`() =
+        test {
+            val scanStateItems = buildScanStateItems(scanStateModelWithThreats)
+
+            assertThat(scanStateItems.filterIsInstance(IconState::class.java).first()).isEqualTo(
+                IconState(
+                    icon = R.drawable.ic_shield_warning_white,
+                    colorResId = R.color.error,
+                    sizeResId = R.dimen.scan_icon_size,
+                    marginResId = R.dimen.scan_icon_margin,
+                    contentDescription = UiStringRes(R.string.scan_state_icon)
+                )
+            )
+        }
+
+    @Test
+    fun `given idle state with threats, when items are built, items contain header for threats found`() = test {
+        val scanStateItems = buildScanStateItems(model = scanStateModelWithThreats)
+
+        assertThat(scanStateItems.filterIsInstance(HeaderState::class.java).first()).isEqualTo(
+            HeaderState(UiStringRes(R.string.scan_idle_threats_found_title))
         )
+    }
 
-        assertThat(scanStateItems.filterIsInstance(ThreatItemState::class.java).first().subHeader)
-            .isEqualTo(subHeader.text)
+    @Test
+    fun `given idle state with threats, when items are built, items contain threats found description`() = test {
+        buildScanStateItems(scanStateModelWithThreats)
+
+        verify(htmlMessageUtils).getHtmlMessageFromStringFormatResId(
+            R.string.scan_idle_with_threats_description,
+            "<b>${threats.size}</b>",
+            "<b>${site.name ?: resourceProvider.getString(R.string.scan_this_site)}</b>",
+            resourceProvider.getString(R.string.scan_here_to_help)
+        )
     }
 
-    /* IDLE - THREATS FOUND STATE */
+    @Test
+    fun `given idle state with threats, when items are built, items contain scan now button`() = test {
```

Comment:
Minor (🔍): Consider fixing the typo in `now`. What you probably want is `again`, isn't it?

Reply:
Fixed in b309ef4